packages/host/app/routes/index.ts

```diff
@@ -87,33 +74,21 @@ export default class Index extends Route<void> {
         cardUrl = new URL(cardPath, window.location.origin).href;
       }
 
-      let cardOrError = await this.fetchCard(cardUrl); // We are fetching the card here to make sure we have the canonical URL for the card
-      if (
-        !cardOrError ||
-        (!isCardInstance(cardOrError) && cardOrError.status === 401)
-      ) {
-        // This is a temporary workaround to fix a strange race condition that happens after user tries to see a
-        // card but has to login first. Immediately after login the route will refresh this.fetchCard(cardUrl) will sometimes fail
-        // with a 401 even when the user has permissions to access the card. There has to be a race condition where this happens before
-        // the realm session is established.
-        // TODO: fix the root cause instead of this hack
-        await new Promise((resolve) => setTimeout(resolve, 500)); // Wait for a bit to so that the realm session is established
-        cardOrError = await this.fetchCard(cardUrl);
-      }
-
-      let canonicalCardUrl = cardOrError?.id;
-
+      // we only get a card to understand its canonical URL so it's ok to fetch
+      // a card that is detached from the store as we only care about it's ID.
+      let canonicalCardUrl: string | undefined;
+      canonicalCardUrl = (
+        await this.store.getInstanceDetachedFromStore(cardUrl)
+      )?.id;
       if (!canonicalCardUrl) {
         // TODO: show a 404 page
         // https://linear.app/cardstack/issue/CS-7364/show-user-a-clear-message-when-they-try-to-access-a-realm-they-cannot
         alert(`Card not found: ${cardUrl}`);
       }
-
```

Comment:
Is this your editor removing these, or a preference, or? Just for curiosity 🤓

Reply:
it was probably me. likely a bunch of updates that I made and then backed out and this remained.